src/devices/Common/Iot/Device/Common/WeatherHelper.cs

```diff
@@ -0,0 +1,102 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Iot.Units;
+using System;
+
+namespace Iot.Device.Common
+{
+    /// <summary>
+    /// Helpers for weather.
+    /// </summary>
+    public class WeatherHelper
+    {
+        /// <summary>
+        /// The heat index (or apparent temperature) is used to measure the amount of discomfort
+        /// during the summer months when heat and humidity often combine to make it feel hotter
+        /// than it actually is. The heat index is usually used for afternoon high temperatures.
+        /// Use the summer simmer index instead for overnight low temperatures.
+        /// </summary>
+        /// <param name="airTemperature">The dry air temperature</param>
+        /// <param name="relativeHumidity">The relative humidity (RH) expressed as a percentage</param>
+        /// <returns>The heat index, also known as the apparent temperature</returns>
+        public static Temperature HeatIndex(Temperature airTemperature, double relativeHumidity)
+        {
+            double tf = airTemperature.Fahrenheit;
+            double rh = relativeHumidity;
+            return Temperature.FromFahrenheit((-42.379)
+                + (2.04901523 * tf)
+                + (10.14333127 * rh)
+                - (0.22475541 * tf * rh)
+                - (6.83783 * Math.Pow(10, -3) * Math.Pow(tf, 2))
```

Comment:
Some of these seem to be repeating (i.e. `Math.Pow(tf, 2)`) - you might want to create local variables to store those

Reply:
The compiler will take care of that, I believe. I'm following formulas I found and are generally accepted. The more I deviate from the standard, the harder it becomes for people to follow.

Reply:
Done